client/stripe-link/index.js

```diff
@@ -52,39 +52,75 @@ const enableStripeLinkPaymentMethod = ( options ) => {
 			  };
 
 		if ( options.complete_shipping() ) {
+			const shippingNames = shippingAddress.name.split( / (.*)/s, 2 );
+			shippingAddress.address.last_name = shippingNames[ 1 ];
+			shippingAddress.address.first_name = shippingNames[ 0 ];
+
 			fillWith( shippingAddress, options.shipping_fields.line1, 'line1' );
 			fillWith( shippingAddress, options.shipping_fields.line2, 'line2' );
 			fillWith( shippingAddress, options.shipping_fields.city, 'city' );
-			fillWith( shippingAddress, options.shipping_fields.state, 'state' );
 			fillWith(
 				shippingAddress,
-				options.shipping_fields.postal_code,
-				'postal_code'
+				options.shipping_fields.country,
+				'country'
 			);
 			fillWith(
 				shippingAddress,
-				options.shipping_fields.country,
-				'country'
+				options.shipping_fields.first_name,
+				'first_name'
+			);
+			fillWith(
+				shippingAddress,
+				options.shipping_fields.last_name,
+				'last_name'
+			);
+			jQuery(
+				'#billing_country, #billing_state, #shipping_country, #shipping_state'
+			).trigger( 'change' );
+			fillWith( shippingAddress, options.shipping_fields.state, 'state' );
+			fillWith(
+				shippingAddress,
+				options.shipping_fields.postal_code,
+				'postal_code'
 			);
 		}
 
 		if ( options.complete_billing() ) {
+			const billingNames = billingAddress.name.split( / (.*)/s, 2 );
+			billingAddress.address.last_name = billingNames[ 1 ];
+			billingAddress.address.first_name = billingNames[ 0 ];
```

Comment:
> If someone like that is paying then billingNames[ 1 ] will return undefined, so I think we should make sure we handle this case with a ?? '':
fixed in [eb6f92f](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2457/commits/eb6f92fc9cd80639dfb496080c1912408228271f) . thank you!

>On a separate (and probably less serious) note people can have multiple first names just as they can have multiple last names. Is there any way we can infer that from the information available here? If not, I think taking the first "word" in a name (like you've already done) is a decent assumption to make — although sometimes titles are included in first names which could make this confusing. However, if we do have access to better information it would probably be best to use that.

At this point Stripe does not share any other information.

Thank you for the review .